sdk/core/Azure.Core/src/Pipeline/LoggingPolicy.cs

```diff
@@ -90,79 +72,32 @@ private async ValueTask ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory
 
             if (isError)
             {
                 s_eventSource.ErrorResponse(message.Response);
-
-                if (!wrapResponseContent && logResponseContent)
-                {
-                    if (textResponse)
-                    {
-                        if (async)
-                        {
-                            await s_eventSource.ErrorResponseContentTextAsync(message.Response, responseTextEncoding!, message.CancellationToken).ConfigureAwait(false);
-                        }
-                        else
-                        {
-                            s_eventSource.ErrorResponseContentText(message.Response, responseTextEncoding!);
-                        }
-                    }
-                    else
-                    {
-                        if (async)
-                        {
-                            await s_eventSource.ErrorResponseContentAsync(message.Response, message.CancellationToken).ConfigureAwait(false);
-                        }
-                        else
-                        {
-                            s_eventSource.ErrorResponseContent(message.Response);
-                        }
-                    }
-                }
+            }
+            else
+            {
+                s_eventSource.Response(message.Response);
             }
 
-            s_eventSource.Response(message.Response);
-
-            if (!wrapResponseContent && logResponseContent)
+            if (wrapResponseContent)
             {
-                if (textResponse)
-                {
-                    if (async)
-                    {
-                        await s_eventSource.ResponseContentTextAsync(message.Response, responseTextEncoding!, message.CancellationToken).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        s_eventSource.ResponseContentText(message.Response, responseTextEncoding!);
-                    }
-                }
-                else
-                {
-                    if (async)
-                    {
-                        await s_eventSource.ResponseContentAsync(message.Response, message.CancellationToken).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        s_eventSource.ResponseContent(message.Response);
-                    }
-                }
+                message.Response.ContentStream = new LoggingStream(message.Response.ClientRequestId, logWrapper, _maxLength, message.Response.ContentStream!, isError, responseTextEncoding);
+            }
+            else
+            {
+                await logWrapper.LogAsync(message.Response.ClientRequestId, isError, message.Response.ContentStream, responseTextEncoding, async).ConfigureAwait(false).EnsureCompleted(async);
             }
 
             var elapsedMilliseconds = (after - before) * 1000 / s_frequency;
```

Comment:
nit: I found frequency to be a bit vague here. After looking in the docs I saw that the frequency is really ticks per second. What about naming variables something like s_ticksPerSecond, endTicks, startTicks?

Reply:
I'll just remove the field, it serves no real purpose.